src/sentry/scim/endpoints/teams.py

```diff
@@ -0,0 +1,279 @@
+
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.endpoints.team_details import TeamSerializer
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.paginator import GenericOffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.team import TeamSCIMSerializer
+from sentry.models import (
+    AuditLogEntryEvent,
+    OrganizationMember,
+    OrganizationMemberTeam,
+    Team,
+    TeamStatus,
+)
+from sentry.signals import team_created
+from sentry.tasks.deletion import delete_team
+from sentry.utils.cursors import SCIMCursor
+
+from .constants import SCIM_400_INVALID_FILTER, SCIM_404_USER_RES, SCIM_API_LIST, GroupPatchOps
+from .utils import OrganizationSCIMTeamPermission, SCIMEndpoint, parse_filter_conditions
+
+delete_logger = logging.getLogger("sentry.deletions.api")
+
+
+CONFLICTING_SLUG_ERROR = "A team with this slug already exists."
+
+
+class SCIMTeamNameSerializer(serializers.Serializer):
+    name = serializers.CharField(max_length=64, required=True, allow_null=False, allow_blank=False)
+
+    def validate(self, attrs):
+        if not (attrs.get("name")):
+            raise serializers.ValidationError("Name is required")
+        return attrs
+
+
+class OrganizationSCIMTeamIndex(SCIMEndpoint, OrganizationEndpoint):
+    permission_classes = (OrganizationSCIMTeamPermission,)
+
+    def get(self, request, organization):
+        try:
+            filter_val = parse_filter_conditions(request.GET.get("filter"))
+        except ValueError:
+            raise ParseError(detail=SCIM_400_INVALID_FILTER)
+
+        if "members" in request.GET.get("excludedAttributes", []):
+            exclude_members = True
+        else:
+            exclude_members = False
+        queryset = Team.objects.filter(
+            organization=organization, status=TeamStatus.VISIBLE
+        ).order_by("slug")
+
+        if filter_val:
+            queryset = queryset.filter(slug=slugify(filter_val))
+
+        def data_fn(offset, limit):
+            return list(queryset[offset : offset + limit])
+
+        def on_results(results):
+            results = serialize(
+                results, None, TeamSCIMSerializer(), exclude_members=exclude_members
+            )
+            return {
+                "schemas": [SCIM_API_LIST],
+                "totalResults": queryset.count(),  # TODO: audit perf
+                "startIndex": int(request.GET.get("startIndex", 1)),  # must be integer
+                "itemsPerPage": len(results),  # what's max?
+                "Resources": results,
+            }
+
+        return self.paginate(
+            request=request,
+            on_results=on_results,
+            paginator=GenericOffsetPaginator(data_fn=data_fn),
+            default_per_page=int(request.GET.get("count", 100)),
+            queryset=queryset,
+            cursor_cls=SCIMCursor,
+        )
+
+    def post(self, request, organization):
```

Comment:
for POST and DELETE there is a lot of duplication with the regular teams route. I will try to rectify these